cci_nepal/pipeline/classification_model/model_save_run/model_save.py

```diff
@@ -0,0 +1,133 @@
+from sklearn.preprocessing import RobustScaler
+from sklearn.multioutput import MultiOutputClassifier
+from sklearn.linear_model import LogisticRegression
+import pickle
+
+# Project libraries
+import cci_nepal
+from cci_nepal.pipeline.classification_model import model_tuning_report as mtr
+from cci_nepal.getters.classification_model import get_real_data as grd
+from cci_nepal.pipeline.classification_model import data_manipulation as dm
+from cci_nepal import config
+
+# %%
+# Set the project directory
+project_dir = cci_nepal.PROJECT_DIR
+
+# %%
+# Read data and feature names
+train = grd.read_train_data()
+val = grd.read_val_data()
+column_names = grd.get_lists(f"{project_dir}/cci_nepal/config/column_names.csv")
+select_features = grd.get_lists(f"{project_dir}/cci_nepal/config/select_features.csv")
+
+# %%
+# Get parameters from config file
+b_features = config["final_model"]["basic_model_features"]
+nb_features = config["final_model"]["non_basic_model_features"]
+lr_solver = config["final_model"]["solver"]
+b_penalty = config["final_model"]["penalty_basic"]
+nb_penalty = config["final_model"]["penalty_non_basic"]
+
+# %%
+# Combine training and validation sets
+train = pd.concat([train, val])
+train = shuffle(train, random_state=1)
+
+# %%
+# Lowercase values
+train = train.applymap(lambda s: s.lower() if type(s) == str else s)
+
+# %%
+# Items, basic and non-basic divide
+nfri_items = column_names[37:]
+basic = nfri_items[0:11]
+non_basic = nfri_items[11:]
+
+# %%
+# Data transformations and feature creation
+dm.transform_sets(train, column_names)
+dm.feature_creation(train)
+
+# %%
+# X and Y split
+X_train = train[select_features]
+y_train = train[nfri_items]
+
+# %%
+# Preferences to binary
+y_train = dm.nfri_preferences_to_binary(y_train)
+
+# %%
+# Split basic /non basic
+y_train_basic = y_train[basic]
+y_train_non_basic = y_train[non_basic]
+
+# %%
+# Define the transformations to be made
+transformer = mtr.col_transformer()
+
+# %%
+# Best performing model
+logr_b = MultiOutputClassifier(LogisticRegression(solver=lr_solver, penalty=b_penalty))
+logr_nb = MultiOutputClassifier(
+    LogisticRegression(solver=lr_solver, penalty=nb_penalty)
+)
+
+# %%
+# Apply column transformer
+X_train = transformer.fit_transform(X_train)
+
+# %%
+# Assign back to dataframes - to have feature names back
+X_train = pd.DataFrame(X_train, columns=list(transformer.get_feature_names_out()))
+
+# %%
+# Reduce to just chosen features for basic and non-basic
+X_train_basic = X_train[b_features].copy()
+X_train_non_basic = X_train[nb_features].copy()
+
+# %%
+# Fit model
+basic_model = logr_b.fit(X_train_basic, y_train_basic)
+# Save model to disk
+filename = f"{project_dir}/outputs/models/final_classification_model_basic.sav"
+pickle.dump(basic_model, open(filename, "wb"))
+
+# Fit model
+non_basic_model = logr_nb.fit(X_train_non_basic, y_train_non_basic)
+# Save model to disk
+filename = f"{project_dir}/outputs/models/final_classification_model_non_basic.sav"
```

Comment:
This failed for me as I didn't have a nested folder `models`. Can we try to check for existing folder and create folder if necessary?

Reply:
Updated like the above suggestion to add a folder called models if not already created - thanks!